File: backend/backend/Core/Models/Response.cs
using System.Net;

namespace backend.Core.Models;

public class Response<TR>
{
    public TR? Data { get; set; }
    public bool IsSuccess { get; set; } = true;
    public string ErrorMessage { get; set; } = string.Empty;
    public HttpStatusCode? StatusCode { get; set; }
}

File: frontend/src/helpers/response.ts
import {HttpStatusCode} from "axios";

export type BaseResponse = {
	isSuccess?: boolean;
	message?: string
}
export type Response<T> = {
	isSuccess?: boolean;
	data?: T;
	errorMessage?: string;
	statusCode?: HttpStatusCode;
}


export type CompanyResponseDto = {
	id?: number;
	createdAt?: Date;
	updatedAt?: Date;
	isActive?: boolean;
	name?: string;
	size?: string;
}

export type JobResponseDto = {
	id?: number;
	createdAt?: Date;
	title?: string;
	description?: string;
	jobLevel?: string;
	companyId?: number;
	companyName?: string;
}
export type CandidatesResponseDto = {
	id?: number;
	firstName?: string;
	lastName?: string;
	email?: string;
	phoneNumber?: string;
	coverLetter?: string;
	resumeUrl?: string;
	jobId?: number;
}